File: frontend/types/payment.ts
export type PaymentStatus = 'paid' | 'pending' | 'failed' | 'refunded';

export interface Payment {
  id: string;
  user_id: string;
  subscription_id: string;
  amount: number;
  payment_method: string;
  transaction_id: string;
  status: PaymentStatus;
  paid_at: string;
  created_at: string;
  customer_name?: string;
  plan_name?: string;
}

File: backend/app/schemas/payment.py
from pydantic import BaseModel
from typing import Optional

class PaymentResponse(BaseModel):
    id: str
    user_id: str
    subscription_id: str
    amount: float
    payment_method: str
    transaction_id: str
    status: str # paid | pending | failed | refunded
    paid_at: Optional[str] = None
    created_at: Optional[str] = None
    customer_name: Optional[str] = None
    plan_name: Optional[str] = None
